packages/snaps-sdk/src/types/methods/notify.ts

```diff
@@ -17,14 +33,47 @@ export enum NotificationType {
  * @property type - The type of notification to display.
  * @property message - The message to display in the notification.
  */
-export type NotifyParams = {
-  type: EnumToUnion<NotificationType>;
-  message: string;
-};
+export type NotifyParams =
+  | {
+      type: EnumToUnion<NotificationType.Native>;
+      message: string;
+    }
+  | {
+      type: EnumToUnion<NotificationType.InApp>;
+      message: string;
+    }
+  | {
+      type: EnumToUnion<NotificationType.InApp>;
+      message: string;
+      detailedView: NotificationComponent;
+      title: string;
+    }
+  | {
+      type: EnumToUnion<NotificationType.InApp>;
+      message: string;
+      detailedView: NotificationComponent;
+      title: string;
+      footerLink: { href: string; text: string };
```

Comment:
If we allow this, we should make sure to validate the link against the phishing list.

Reply:
Yeah, it is: [ ](https://github.com/MetaMask/snaps/pull/2706/files#diff-1137a455e041bffaa796550025741586b2eb522d5e6daef7907fc111237f22d7R264)